packages/tools/src/utilities/segmentation/thresholdVolumeByRange.ts

```diff
@@ -1,34 +1,45 @@
-import type { Types } from '@cornerstonejs/core';
+import { Types } from '@cornerstonejs/core';
+import { utilities as csUtils } from '@cornerstonejs/core';
 
 import { pointInShapeCallback } from '../../utilities';
 import { triggerSegmentationDataModified } from '../../stateManagement/segmentation/triggerSegmentationEvents';
 import { BoundsIJK } from '../../types';
+import getBoundingBoxAroundShape from '../boundingBox/getBoundingBoxAroundShape';
 
 export type ThresholdRangeOptions = {
-  upper: number;
-  lower: number;
-  boundsIJK: BoundsIJK;
   overwrite: boolean;
+  boundsIJK: BoundsIJK;
+  overlapType?: number;
+};
+
+export type ThresholdInformation = {
+  volume: Types.IImageVolume;
+  lower: number;
+  upper: number;
 };
 
 /**
  * It thresholds a segmentation volume based on a set of threshold values with
- * respect to reference volumes.
+ * respect to a list of volumes and respective threshold ranges.
  * @param segmentationVolume - the segmentation volume to be modified
- * @param referenceVolumes - the reference volumes to be used for thresholding
+ * @param thresholdVolumeInformation - array of objects containing volume data
+ * and a range (lower and upper values) to threshold
  * @param options - the options for thresholding
  * @returns segmented volume
  */
 function thresholdVolumeByRange(
   segmentationVolume: Types.IImageVolume,
-  referenceVolume: Types.IImageVolume,
+  thresholdVolumeInformation: ThresholdInformation[],
   options: ThresholdRangeOptions
 ): Types.IImageVolume {
   const { scalarData, imageData: segmentationImageData } = segmentationVolume;
-  const { overwrite, boundsIJK, upper, lower } = options;
 
-  const { imageData } = referenceVolume;
-  const referenceValues = imageData.getPointData().getScalars().getData();
+  const { overwrite, boundsIJK } = options;
+  let overlapType = options.overlapType;
+
+  if (!overlapType) {
+    overlapType = 0; // default is any overlap
+  }
```

Comment:
You can do this in the function argument by saying `overlayType=0` to have it default to 0

Reply:
Done.
const overlapType = options?.overlapType || 0;